src/ir/analysis/derive.rs

```diff
@@ -549,8 +611,8 @@ impl DeriveTrait {
     fn can_derive_pointer(&self) -> CanDerive {
         match self {
             DeriveTrait::Default => {
-                trace!("    pointer cannot derive Default");
-                CanDerive::No
+                trace!("    pointer cannot derive Default, but it may be implemented");
```

Comment:
Hmm, how can it be implemented for a pointer?

Reply:
Literally speaking we can't implement anything on pointers, but

```
struct Test {
    field: *mut i32,
}

impl Default for Test {
    fn default() -> Self {
        unsafe { ::std::mem::zeroed() }
    }
}
```

does seem to compile for composite types that contain pointers.